dotCMS/src/main/java/com/dotmarketing/business/UserAPIImpl.java

```diff
@@ -51,124 +53,137 @@
 		}
-		User u = uf.loadByUserByEmail(email);
+		
+		final User u = userFactory.loadByUserByEmail(email);
 		if(!UtilMethods.isSet(u)){
 			throw new com.dotmarketing.business.NoSuchUserException("No user found with passed in email");
 		}
-		if(perAPI.doesUserHavePermission(upAPI.getUserProxy(u,APILocator.getUserAPI().getSystemUser(), false), PermissionAPI.PERMISSION_READ, user, respectFrontEndRoles)){
+		if(permissionAPI.doesUserHavePermission(userProxyAPI.getUserProxy(u,APILocator.getUserAPI().getSystemUser(), false), PermissionAPI.PERMISSION_READ, user, respectFrontEndRoles)){
 			return u;
 		}else{
 			throw new DotSecurityException("The User being passed in doesn't have permission to requested User");
 		}
 	}
 
+	@CloseDBIfOpened
 	@Override
-	public String encryptUserId(String userId) throws DotStateException{
+	public String encryptUserId(final String userId) throws DotStateException{
 		try{
 			return UserManagerUtil.encryptUserId(userId);
 		}catch (Exception e) {
 			throw new DotStateException("Unable to encrypt userID : ", e);
 		}
 	}
 
+	@CloseDBIfOpened
 	@Override
 	public long getCountUsersByName(String filter) throws DotDataException {
-		return uf.getCountUsersByName(filter);
+		return userFactory.getCountUsersByName(filter);
 	}
 
+	@CloseDBIfOpened
 	@Override
 	public List<User> getUsersByName(String filter, int start,int limit, User user, boolean respectFrontEndRoles) throws DotDataException {
-		return uf.getUsersByName(filter, start, limit);
+		return userFactory.getUsersByName(filter, start, limit);
 	}
 
+	@WrapInTransaction
 	@Override
 	public User createUser(String userId, String email) throws DotDataException, DuplicateUserException {
-		return uf.createUser(userId, email);
+		return userFactory.createUser(userId, email);
 	}
 
+	@CloseDBIfOpened
 	@Override
 	public User getDefaultUser() throws DotDataException {
 		try {
-			return uf.loadDefaultUser();
+			return userFactory.loadDefaultUser();
 		} catch (Exception e) {
 			throw new DotDataException("getting default user user failed", e);
 		}
 	}
+
     @Override
 	public User getSystemUser() throws DotDataException {
       if(this.systemUser==null){
         this.systemUser=_getSystemUser();
 	   }
 	   return this.systemUser;
 	}
-	
-	
-	
 
+	@WrapInTransaction
 	private User _getSystemUser() throws DotDataException {
 		User user = null;
 		RoleAPI roleAPI = com.dotmarketing.business.APILocator.getRoleAPI();
 		Role cmsAdminRole = roleAPI.loadCMSAdminRole();
 		try {
-			user = uf.loadUserById(SYSTEM_USER_ID);
+			user = userFactory.loadUserById(SYSTEM_USER_ID);
 		} catch (NoSuchUserException e) {
 			user = createUser("system", "system@dotcmsfakeemail.org");
 			user.setUserId(SYSTEM_USER_ID);
 			user.setFirstName("system user");
 			user.setLastName("system user");
 			user.setCreateDate(new java.util.Date());
 			user.setCompanyId(PublicCompanyFactory.getDefaultCompanyId());

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "PublicCompanyFactory"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)